### DISCUSSION TITLE
Add exit node routing into the UI or options

### DISCUSSION THREAD

**Discussion by mfratto** [hassio-addons/app-tailscale#59] [Ideas]
Great add-on. Works well. Thanks for the effort. One suggestion. Adding the ability to add exit node subnets to the installation. For example, my home network is on 192.168.1.x and my camera network is on 192.168.2.x. Using tailscale, I can add the .2 subnet as a CLI command but there doesn't seem to be a way to do so in this add-on.

**Comment by kgleason** [comment]
It would also be nice to have the option of disabling either the exit node or the subnet router.

**Comment by GpMissi** [comment]
I approve. It is just what I am looking for

**Comment by carlos16196** [comment]
Yes can we please add option to set subnet routes ?

**Comment by lmagyar** [comment]
Core HA team insists on dealing with only the Supervisor managed interfaces (really understandable), so I see very little chance that subnet routing will be configurable.

I've added this functionality to my fork, you can install and test it from here: https://github.com/lmagyar/homeassistant-addon-tailscale

It advertises the subnets and configures MSS clamping on their interfaces as required by [TS documentation](https://tailscale.com/kb/1214/site-to-site/). All the other network configuration is your responsibility. It also protects local subnets to be redirected toward your tailnet to protect against soft-bricking your device (you can accidentally advertise eg. 192.168.1.0/24 on multiple sites, but TS will think this is load balancing and/or failover, and your HA will lose all network access, so this add-on tries to prevent it, though it makes load balancing and/or failover impossible through this add-on).

Note: At first step, don't turn off `snat-subnet-routes`, only when you really need this.

**Comment by lmagyar** [comment]
Exit node config is already there.
